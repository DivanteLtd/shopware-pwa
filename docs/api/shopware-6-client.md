<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@shopware-pwa/shopware-6-client](./shopware-6-client.md)

## shopware-6-client package

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [ClientSettings](./shopware-6-client.clientsettings.md) | <b><i>(BETA)</i></b> |

## Variables

|  Variable | Description |
|  --- | --- |
|  [config](./shopware-6-client.config.md) | <b><i>(BETA)</i></b> |

