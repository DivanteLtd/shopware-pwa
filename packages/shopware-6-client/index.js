"use strict";

if (process.env.NODE_ENV === "production") {
  module.exports = require("./dist/shopware-6-client.cjs.prod.js");
} else {
  module.exports = require("./dist/shopware-6-client.cjs.js");
}
