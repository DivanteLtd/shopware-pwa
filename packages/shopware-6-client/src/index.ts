export * from "./index.inner";

// Explicitly import offline overrides and alias them

// Product Listing
import { getCategoryProducts as offlineGetCategoryProducts } from "./offline-services/productService";

const getCategoryProducts = offlineGetCategoryProducts;

// Product Search
import {
  searchProducts as offlineSearchProducts,
  searchSuggestedProducts as offlineSearchSuggestedProducts,
} from "./offline-services/searchService";

const searchProducts = offlineSearchProducts;
const searchSuggestedProducts = offlineSearchSuggestedProducts;

// Re-export them with original name, because local modules have priority in bundler
export { searchProducts, searchSuggestedProducts, getCategoryProducts };
