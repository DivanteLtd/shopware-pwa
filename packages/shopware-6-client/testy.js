const { getUserCountry, setup } = require("@shopware-pwa/shopware-6-client");

!(async () => {
  setup({
    endpoint: "http://localhost:8000",
    accessToken: "SWSCMGPZM1LVCK5QOHLBSMLAUQ",
  });
  try {
    const response = await getUserCountry("012e32ef96674dd584d71d993fb1f458");
    console.info(response);
  } catch (error) {
    console.warn("error", error);
  }
})();
