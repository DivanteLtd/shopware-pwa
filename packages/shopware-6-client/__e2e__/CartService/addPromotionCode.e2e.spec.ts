import { update, addPromotionCode } from "@shopware-pwa/shopware-6-client";
describe("shopware-6-client - E2E - CartService - addCartItemQuantity", () => {
  beforeEach(() => {
    update({ contextToken: "" })
  });

  it("should add promotion code for products in cart", async() => {
    const result = await addPromotionCode("divante");
    result.token = "mockedToken";
    expect(result).toMatchSnapshot();
  });
  
  it("should returns error when adding non-existing promotion code", async () => {
    try {
      await addPromotionCode("que");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch(e) {
      expect(e).toMatchSnapshot();
    }
  });

  it("should returns error when adding empty string as promotion code", async () => {
    try {
      await addPromotionCode("que");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch(e) {
      expect(e).toMatchSnapshot();
    }
  });
});