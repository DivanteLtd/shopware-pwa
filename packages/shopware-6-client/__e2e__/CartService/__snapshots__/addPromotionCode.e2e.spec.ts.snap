// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CartService - addCartItemQuantity should add promotion code for products in cart 1`] = `
Object {
  "deliveries": Array [],
  "errors": Array [],
  "extensions": Array [],
  "lineItems": Array [],
  "modified": false,
  "name": "8a243080f92e4c719546314b577cf82b",
  "price": Object {
    "calculatedTaxes": Array [],
    "extensions": Array [],
    "netPrice": 0,
    "positionPrice": 0,
    "taxRules": Array [],
    "taxStatus": "gross",
    "totalPrice": 0,
  },
  "token": "mockedToken",
  "transactions": Array [
    Object {
      "amount": Object {
        "calculatedTaxes": Array [],
        "extensions": Array [],
        "quantity": 1,
        "referencePrice": null,
        "taxRules": Array [],
        "totalPrice": 0,
        "unitPrice": 0,
      },
      "extensions": Array [],
      "paymentMethodId": "cad2356ee68c473ca7da517d0ad32de1",
    },
  ],
}
`;

exports[`shopware-6-client - E2E - CartService - addCartItemQuantity should returns error when adding empty string as promotion code 1`] = `
[Error: [2mexpect([22m[31mreceived[39m[2m).[22mtoEqual[2m([22m[32mexpected[39m[2m) // deep equality[22m

Expected: [32m"[7mshould[27m throw an error"[39m
Received: [31m"[7mdidn't[27m throw an error"[39m]
`;

exports[`shopware-6-client - E2E - CartService - addCartItemQuantity should returns error when adding non-existing promotion code 1`] = `
[Error: [2mexpect([22m[31mreceived[39m[2m).[22mtoEqual[2m([22m[32mexpected[39m[2m) // deep equality[22m

Expected: [32m"[7mshould[27m throw an error"[39m
Received: [31m"[7mdidn't[27m throw an error"[39m]
`;
